src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -604,6 +668,27 @@ private void EnterpriseAndHyperVToAzure(EnableProtectionInput input)
             providerSettings.TargetProximityPlacementGroupId = this.RecoveryProximityPlacementGroupId;
             providerSettings.TargetAvailabilityZone = this.RecoveryAvailabilityZone;
             providerSettings.UseManagedDisks = this.UseManagedDisk;
+            providerSettings.TargetAvailabilitySetId = this.RecoveryAvailabilitySetId;
+            providerSettings.TargetVmSize = this.Size;
+            providerSettings.SqlServerLicenseType = this.SqlServerLicenseType;
+            providerSettings.TargetVmTags = this.RecoveryVmTag;
+            providerSettings.TargetNicTags = this.RecoveryNicTag;
+
+            if(this.DiskTag != null || this.DiskTag.Count > 0)
+            {
+                if(this.UseManagedDisk == Constants.False)
+                {
+                    throw new PSArgumentException(
+                                string.Format(
+                                    Resources.DiskTagCannotBeSet,
+                                    this.DiskTag,
+                                    this.UseManagedDisk));
+                }
+                else
+                {
+                    providerSettings.TargetManagedDiskTags = this.DiskTag;
```

Comment:
>providerSettings.TargetManagedDiskTags = this.DiskTag [](start = 20, length = 53)

we can assign irrespective of the check like

providerSettings.TargetManagedDiskTags = this.DiskTag;
if(this.UseManagedDisk == Constants.False && (this.DiskTag != null || this.DiskTag.Count > 0)
{
                    throw new PSArgumentException(
                                string.Format(
                                    Resources.DiskTagCannotBeSet,
                                    this.DiskTag,
                                    this.UseManagedDisk));
                }
or u can move the error above